### ISSUE TITLE
Multiple calls to ObjectMapper#canSerialize(Object.class) returns different values.

### ISSUE THREAD

**Issue by flexfrank** [FasterXML/jackson-databind#703]
In 2.5.1, the first call to  `ObjectMapper#canSerialize(Object.class)` returns false.
However, it returns true from the second call.
It seems to be cause by cache mechanism in `SerializerProvider` class.

Sample code:
```Java
ObjectMapper mapper = new ObjectMapper();
System.out.println(mapper.canSerialize(Object.class));
System.out.println(mapper.canSerialize(Object.class));
System.out.println(mapper.canSerialize(Object.class));
```
output:
```
false
true
true
```

**Comment by flozano** [comment]
Reproduced here

**Comment by flozano** [comment]
In SerializerProvider:
```java
    protected JsonSerializer<Object> _findExplicitUntypedSerializer(Class<?> runtimeType)
        throws JsonMappingException
    {        
        // Fast lookup from local lookup thingy works?
        JsonSerializer<Object> ser = _knownSerializers.untypedValueSerializer(runtimeType);
        if (ser == null) {
            // If not, maybe shared map already has it?
            ser = _serializerCache.untypedValueSerializer(runtimeType);
            if (ser == null) {
                ser = _createAndCacheUntypedSerializer(runtimeType);
                /* 18-Sep-2014, tatu: This is unfortunate patch over related change
                 *    that pushes creation of "unknown type" serializer deeper down
                 *    in BeanSerializerFactory; as a result, we need to "undo" creation
                 *    here.
                 */
                if (isUnknownTypeSerializer(ser)) {
                    return null;
                }
            }
        }
        return ser;
    }
```

In first execution of:
```java
        JsonSerializer<Object> ser = _knownSerializers.untypedValueSerializer(runtimeType);
```
`ser==null`, whereas in subsequent executions, `ser` is an instance of `UnknownSerializer`.

I don't know if this is expected or not... but, if this is OK, it could be quickly fixed by moving this:
```java
                if (isUnknownTypeSerializer(ser)) {
                    return null;
                }
```
out of the first if() clause.

**Comment by flozano** [comment]
Proposed fix in https://github.com/FasterXML/jackson-databind/pull/704

**Comment by cowtowncoder** [comment]
@flozano Thank you once again for the fix: I added a test and backported in 2.5 branch (to be included in 2.5.2)

**Comment by flozano** [comment]
np, thanks for making jackson!
